save-backend/src/main/kotlin/com/saveourtool/save/backend/service/LnkOrganizationTestSuiteService.kt

```diff
@@ -0,0 +1,91 @@
+
+package com.saveourtool.save.backend.service
+
+import com.saveourtool.save.backend.repository.LnkOrganizationTestSuiteRepository
+import com.saveourtool.save.domain.Role
+import com.saveourtool.save.entities.*
+import com.saveourtool.save.permission.Rights
+import org.springframework.stereotype.Service
+import kotlin.NoSuchElementException
+
+/**
+ * Service of [LnkOrganizationTestSuite]
+ */
+@Service
+class LnkOrganizationTestSuiteService(
+    private val lnkOrganizationTestSuiteRepository: LnkOrganizationTestSuiteRepository,
+) {
+    /**
+     * @param organization
+     * @return all [TestSuite]s with rights for [organization]
+     */
+    fun getAllTestSuitesByOrganization(organization: Organization) =
+            lnkOrganizationTestSuiteRepository.findByOrganization(organization)
+                .map {
+                    it.testSuite
+                }
+
+    /**
+     * @param organization
+     * @param testSuite
+     * @return [LnkOrganizationTestSuite] by [organization] and [testSuite]
+     */
+    fun findByOrganizationAndTestSuite(organization: Organization, testSuite: TestSuite) =
+            lnkOrganizationTestSuiteRepository.findByOrganizationAndTestSuite(organization, testSuite)
+
+    /**
+     * Set [rights] of [organization] over [testSuite].
+     *
+     * @throws IllegalStateException if [rights] is [Role.NONE]
+     */
+    @Suppress("KDOC_WITHOUT_PARAM_TAG", "UnsafeCallOnNullableType")
+    fun setRights(organization: Organization, testSuite: TestSuite, rights: Rights) {
+        if (rights == Rights.NONE) {
+            throw IllegalStateException("NONE rights should not be present in database!")
+        }
+        val lnkOrganizationTestSuite = findByOrganizationAndTestSuite(organization, testSuite)
+            ?.apply { this.rights = rights }
+            ?: LnkOrganizationTestSuite(organization, testSuite, rights)
+        lnkOrganizationTestSuiteRepository.save(lnkOrganizationTestSuite)
+    }
+
+    /**
+     * Set [rights] of organization with [organizationId] over test suite with [testSuiteId].
+     *
+     * @throws IllegalStateException if [rights] is [Role.NONE]
+     */
+    @Suppress("KDOC_WITHOUT_PARAM_TAG", "UnsafeCallOnNullableType")
+    fun setRightsByIds(testSuiteId: Long, organizationId: Long, rights: Rights) {
+        if (rights == Rights.NONE) {
+            throw IllegalStateException("NONE rights should not be present in database!")
+        }
+        lnkOrganizationTestSuiteRepository.findByOrganizationIdAndTestSuiteId(organizationId, testSuiteId)
+            ?.apply { this.rights = rights }
+            ?.let { lnkOrganizationTestSuiteRepository.save(it) }
+            ?: lnkOrganizationTestSuiteRepository.save(organizationId, testSuiteId, rights.toString())
+    }
+
+    /**
+     * @param organization
+     * @param testSuite
+     * @return [Rights] of [organization] over [testSuite]
+     */
+    fun getRights(organization: Organization, testSuite: TestSuite) =
```

Comment:
I don't think this method is needed. Let's create a LnkOrganizationTestSuiteDTO and we will always return it to the front and take Rights from it

Reply:
I think it should be done right away. If we create a new entity, then we immediately create an DTO for it and use only the DTO in the controller.

Reply:
Ok, but I do not want it to be called so. All the links should add some info to existing entities, but not be an entity with dto itself.

Reply:
Added Dto